packages/logger/src/logger/DiscordTicketTransport.ts

```diff
@@ -56,106 +50,90 @@ export const isDiscordTicketInfo = (info: unknown): info is DiscordTicketInfo =>
-
-        if (!this.client.isReady()) await this.login(); // Log in if not yet established the connection.
-
-        // Get and verify requested Discord channel to post.
-        const channelId = this.channelIds[info.discordTicketChannel];
-        const channel = await this.client.channels.fetch(channelId);
-        if (channel === null) throw new Error(`Discord channel ${channelId} not available!`);
-        if (!channel.isTextBased()) throw new Error(`Invalid type for Discord channel ${channelId}!`);
-
-        // Prepend the $ticket command and concatenate message title and content separated by newline.
-        // Also remove anchor text from links and truncate the message to Discord's max character limit.
-        const header = `$ticket ${info.message}\n`;
-        const content = this.truncateMessage(
-          removeAnchorTextFromLinks(info.mrkdwn),
-          DISCORD_MAX_CHAR_LIMIT - header.length
-        );
-        const message = header + content;
+      // Callback and errors are handled within base class.
+      await super.log(info, callback);
+    } else {
+      // Signal we're done here.
+      callback();
+    }
+  }
 
-        // Add the message to the queue and process it.
-        this.logQueue.push({ channel, message });
-        await this.executeLogQueue();
+  // Get Discord channel by its id and cache fetched ids.
+  private async getChannel(channelId: string, logEntityKey: Key): Promise<TextBasedChannel> {
+    const cachedChannel = this.channels.get(channelId);
+    if (cachedChannel !== undefined) return cachedChannel;
 
-        this.enqueuedLogCounter--; // Decrement counter for the isFlushed getter when done.
-      } catch (error) {
-        this.enqueuedLogCounter--; // Decrement the counter for the isFlushed getter when catching an error.
-        return callback(new TransportError("Discord Ticket", error, info));
-      }
-    }
+    const channel = await this.client.channels.fetch(channelId);
+    if (channel === null)
+      throw new Error(`Discord channel ${channelId} not available, ${this.kind} entity, id=${logEntityKey.id}`);
+    if (!channel.isTextBased())
+      throw new Error(`Invalid type for Discord channel ${channelId}, ${this.kind} entity, id=${logEntityKey.id}`);
 
-    callback();
+    this.channels.set(channelId, channel);
+    return channel;
+  }
+
+  // Logs queue element when processing persistent storage.
+  async logQueueElement(info: Record<string, unknown>, logEntityKey: Key): Promise<void> {
+    if (!isDiscordTicketInfo(info)) throw new Error(`Invalid info in ${this.kind} entity, id=${logEntityKey.id}`);
+
+    // Check if the channel ID is configured.
+    if (!(info.discordTicketChannel in this.channelIds))
+      throw new Error(
+        `Missing channel ID for ${info.discordTicketChannel}, ${this.kind} entity, id=${logEntityKey.id}`
+      );
+
+    if (!this.client.isReady()) await this.login(); // Log in if not yet established the connection.
```

Comment:
Is it possible to move this the constructor so we don't need to check multiple times? It might improve the performance of the logger.